CreateDatabase.sql

```diff
@@ -9,73 +9,177 @@ use airport;
 DROP TABLE IF EXISTS Airport;
-# CREATE TABLE
+CREATE TABLE Airport
+	(ICAO		VARCHAR(4),
+    airportName	VARCHAR(60) NOT NULL,
+    cityName	VARCHAR(85)	NOT NULL,
+    PRIMARY KEY(ICAO),
+    FOREIGN KEY(cityName) REFERENCES City(cityName) ON DELETE CASCADE
+    );
 
 # Gate
 DROP TABLE IF EXISTS Gate;
-# CREATE TABLE
-
-# Ticket
-DROP TABLE IF EXISTS Ticket;
-# CREATE TABLE
-
-# Ticket Price
-DROP TABLE IF EXISTS TicketPrice;
-# CREATE TABLE
-
-# Nationality
-DROP TABLE IF EXISTS Nationality;
-# CREATE TABLE
-
-# Passenger
-DROP TABLE IF EXISTS Passenger;
-# CREATE TABLE
-
-# Crew
-DROP TABLE IF EXISTS Crew;
-# CREATE TABLE
-CREATE TABLE Crew 
-	(crewID INT PRIMARY KEY AUTO_INCREMENT,
-    firstName VARCHAR(50),
-    middleName VARCHAR(100),
-    lastName VARCHAR(50),
-    birthDate DATE,
-    crewRole VARCHAR(50));
-
-# CrewFlight
-DROP TABLE IF EXISTS CrewFlight;
-# CREATE TABLE
-
-# Flight
-DROP TABLE IF EXISTS Flight;
-# CREATE TABLE
+CREATE TABLE Gate
+	(gateID	VARCHAR(4),
+    ICAO	VARCHAR(4),
+    PRIMARY KEY(gateID, ICAO),
+    FOREIGN KEY(ICAO) REFERENCES Airport(ICAO) ON DELETE CASCADE
+	);
 
 # Aircraft Model
 DROP TABLE IF EXISTS AircraftModel;
-# CREATE TABLE
+CREATE TABLE AircraftModel
+	(modelName			VARCHAR(50),
+    manufacturer		VARCHAR(50),
+    seats				INT NOT NULL,
+    licenseDurationDays	INT NOT NULL,
+    PRIMARY KEY(modelName)
+    );
 
 # Aircraft Instance
 DROP TABLE IF EXISTS AircraftInstance;
-# CREATE TABLE
-
-# License
-DROP TABLE IF EXISTS License;
-# CREATE TABLE
+CREATE TABLE AircraftInstance
+	(aircraftReg	VARCHAR(10),
+    productionYear	YEAR NOT NULL,
+    modelName		VARCHAR(50) NOT NULL,
+    PRIMARY KEY(aircraftReg),
+    FOREIGN KEY(modelName) REFERENCES AircraftModel(modelName) ON DELETE CASCADE
+    );
+    
+# Flight
+DROP TABLE IF EXISTS Flight;
+CREATE TABLE Flight
+	(flightID 				INT AUTO_INCREMENT,
+    arrivalDateTimeUTC 		DATETIME NOT NULL,
+    departureDateTimeUTC 	DATETIME NOT NULL,
+    aircraftReg 			VARCHAR(8) NOT NULL,
+    arrivalGateID 			VARCHAR(5) NOT NULL,
+    departureGateID			VARCHAR(5) NOT NULL,
+    arrivalGateAirport		VARCHAR(5) NOT NULL,
+    departureGateAirport	VARCHAR(5) NOT NULL,
+    PRIMARY KEY(flightID),
+    FOREIGN KEY(aircraftReg) REFERENCES AircraftInstance(aircraftReg) ON DELETE CASCADE,
+    FOREIGN KEY(arrivalGateID) REFERENCES Gate(gateID) ON DELETE CASCADE,
+    FOREIGN KEY(departureGateID) REFERENCES Gate(gateID) ON DELETE CASCADE,
+    FOREIGN KEY(arrivalGateAirport) REFERENCES Gate(ICAO) ON DELETE CASCADE,
+    FOREIGN KEY(departureGateAirport) REFERENCES Gate(ICAO) ON DELETE CASCADE
+    );
 
 # Pilot
 DROP TABLE IF EXISTS Pilot;
-# CREATE TABLE
+CREATE TABLE Pilot
+	(pilotID	INT AUTO_INCREMENT,
+    firstName 	VARCHAR(50) NOT NULL,
+    middleName 	VARCHAR(100),
+    lastName 	VARCHAR(50) NOT NULL,
+    birthDate 	DATE NOT NULL,
+    PRIMARY KEY(pilotID)
+    );
+
+# License
+DROP TABLE IF EXISTS License;
+CREATE TABLE License
+	(pilotID			INT,
```

Comment:
Should this be AUTO_INCREMENT?

Reply:
It's generated in the Pilot table and used here as a foreign key, so no.